openbas-api/src/main/java/io/openbas/service/ReportService.java

```diff
@@ -50,6 +53,33 @@ public Report updateReport(@NotNull final Report report, @NotNull final ReportIn
         return this.reportRepository.save(report);
     }
 
+    public List<ReportInjectComment> updateReportInjectComment(@NotNull final Report report, @NotNull final Inject inject, @NotNull final ReportInjectCommentInput input){
+        Optional<ReportInjectComment> reportInjectComment = findReportInjectComment(report.getId(), inject.getId());
+        ReportInjectComment injectComment;
+        if (reportInjectComment.isPresent()) {
+            injectComment = reportInjectComment.get();
+            injectComment.setComment(input.getComment());
+        } else {
+            injectComment = new ReportInjectComment();
+            injectComment.setInject(inject);
+            injectComment.setReport(report);
+            injectComment.setComment(input.getComment());
+            report.getReportInjectsComments().add(injectComment);
+        }
+        this.reportRepository.save(report);
+        return report.getReportInjectsComments();
+    }
+
+    private Optional<ReportInjectComment> findReportInjectComment(String reportId, String injectId) {
+        String jpql = "SELECT r FROM ReportInjectComment r WHERE r.report.id = :reportId AND r.inject.id = :injectId";
```

Comment:
Could be better to have this directly in the ReportRepository

    @Query(value = "SELECT r FROM ReportInjectComment r WHERE r.report.id = :reportId AND r.inject.id = :injectId", nativeQuery = true)
    Optional<ReportInjectComment> method(@Param("PARAM")...);